### PR TITLE
Content

### PR DESCRIPTION
**Pull request by GordanM**

### CODE REVIEW

File: conf/layer.conf [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+# We have a conf and classes directory, append to BBPATH
+BBPATH .= ":${LAYERDIR}"
+
+# We have a recipes directory, add to BBFILES
+BBFILES += "${LAYERDIR}/recipes-*/*/*.bb ${LAYERDIR}/recipes-*/*/*.bbappend"
+
+BBFILE_COLLECTIONS += "template-layer"
+BBFILE_PATTERN_template-layer := "^${LAYERDIR}/"
+
+BBFILE_PRIORITY_template-layer = "7"
```
Comment (rpannek) [inline conf/layer.conf]: Perhaps a short comment on what that magic number means would be nice?

File: recipes-example/template-library/template-library.bb [hunk 2]
```diff
@@ -0,0 +1,25 @@
+#
+#   Copyright (C) 2017 Pelagicore AB
```
Comment (rpannek) [inline recipes-example/template-library/template-library.bb]: I think you missed the SPDX line?

File: recipes-example/template-library/template-library.bb [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+
+DESCRIPTION = "A template C/C++ library"
+HOMEPAGE = "https://github.com/Pelagicore/template-library"
+LICENSE = "MPL-2.0"
+LIC_FILES_CHKSUM = "file://LICENSE;md5=815ca599c9df247a0c7f619bab123dad"
+
+PR = "r0"
+PV = "1.0+git${SRCREV}"
+
+SRC_URI = "gitsm://github.com/Pelagicore/template-library.git;protocol=https;branch=master"
```
Comment (rpannek) [inline recipes-example/template-library/template-library.bb]: I don't think that the template-library has any git submodules, please replace the uri from gitsm to git.

File: recipes-example/template-library/template-library.bb [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+
+SRCREV = "29d6dbd859d725ee6d10686ee5449069412de2c6"
+
+inherit cmake pkgconfig
+
+S = "${WORKDIR}/git/"
+
+PACKAGES = "${PN} ${PN}-dbg ${PN}-dev"
```
Comment (rpannek) [inline recipes-example/template-library/template-library.bb]: Perhaps it's just me but I don't really like to have more than one of those in one line, you can overlook them so easily.

File: recipes-example/template-service/template-service.bb [hunk 5]
```diff
@@ -0,0 +1,23 @@
+#
+#   Copyright (C) 2017 Pelagicore AB
```
Comment (rpannek) [inline recipes-example/template-service/template-service.bb]: Here the SPDX line is missing too.

File: recipes-example/template-service/template-service.bb [hunk 6]
```diff
@@ -0,0 +1,23 @@
...
+
+DESCRIPTION = "A template C/C++ service"
+HOMEPAGE = "https://github.com/GordanM/template-service.git"
+LICENSE = "MPL-2.0"
+LIC_FILES_CHKSUM = "file://LICENSE;md5=815ca599c9df247a0c7f619bab123dad"
+
+PR = "r0"
+PV = "1.0+git${SRCREV}"
+
+SRC_URI = "gitsm://github.com/GordanM/template-service.git;protocol=https;branch=topic/template-library"
```
Comment (rpannek) [inline recipes-example/template-service/template-service.bb]: I don't think this uses git submodules either, you should replace it with git://

**Comment by GordanM** [comment]
@rpannek addressed your concerns!

**Pull request merged**